plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-system-status-v2-controller.php

```diff
@@ -1013,93 +1066,107 @@ public function get_dropins_mu_plugins() {
+					$theme_file = get_stylesheet_directory() . '/' . $file;
+				} elseif ( file_exists( get_stylesheet_directory() . '/' . WC()->template_path() . $file ) ) {
+					$theme_file = get_stylesheet_directory() . '/' . WC()->template_path() . $file;
+				} elseif ( file_exists( get_template_directory() . '/' . $file ) ) {
+					$theme_file = get_template_directory() . '/' . $file;
+				} elseif ( file_exists( get_template_directory() . '/' . WC()->template_path() . $file ) ) {
+					$theme_file = get_template_directory() . '/' . WC()->template_path() . $file;
+				} else {
+					$theme_file = false;
 				}
 
-				$core_version  = WC_Admin_Status::get_file_version( WC()->plugin_path() . '/templates/' . $core_file );
-				$theme_version = WC_Admin_Status::get_file_version( $theme_file );
-				if ( $core_version && ( empty( $theme_version ) || version_compare( $theme_version, $core_version, '<' ) ) ) {
-					if ( ! $outdated_templates ) {
-						$outdated_templates = true;
+				if ( ! empty( $theme_file ) ) {
+					$core_file = $file;
+
+					// Update *-product_<cat|tag> template name before searching in core.
+					if ( false !== strpos( $core_file, '-product_cat' ) || false !== strpos( $core_file, '-product_tag' ) ) {
+						$core_file = str_replace( '_', '-', $core_file );
+					}
+
+					$core_version  = WC_Admin_Status::get_file_version( WC()->plugin_path() . '/templates/' . $core_file );
+					$theme_version = WC_Admin_Status::get_file_version( $theme_file );
+					if ( $core_version && ( empty( $theme_version ) || version_compare( $theme_version, $core_version, '<' ) ) ) {
+						if ( ! $outdated_templates ) {
+							$outdated_templates = true;
+						}
 					}
+					$override_files[] = array(
+						'file'         => str_replace( WP_CONTENT_DIR . '/themes/', '', $theme_file ),
+						'version'      => $theme_version,
+						'core_version' => $core_version,
+					);
 				}
-				$override_files[] = array(
-					'file'         => str_replace( WP_CONTENT_DIR . '/themes/', '', $theme_file ),
-					'version'      => $theme_version,
-					'core_version' => $core_version,
-				);
 			}
+
+			$active_theme_info = array(
+				'name'                    => $active_theme->name,
+				'version'                 => $active_theme->version,
+				'version_latest'          => WC_Admin_Status::get_latest_theme_version( $active_theme ),
+				'author_url'              => esc_url_raw( $active_theme->{'Author URI'} ),
+				'is_child_theme'          => is_child_theme(),
+				'has_woocommerce_support' => current_theme_supports( 'woocommerce' ),
+				'has_woocommerce_file'    => ( file_exists( get_stylesheet_directory() . '/woocommerce.php' ) || file_exists( get_template_directory() . '/woocommerce.php' ) ),
+				'has_outdated_templates'  => $outdated_templates,
+				'overrides'               => $override_files,
+			);
+
+			$theme_info = array_merge( $active_theme_info, $parent_theme_info );
+			set_transient( 'wc_system_status_theme_info', $theme_info );
```

Comment:
Some people might change their theme without triggering an update process. There's probably not a good way to detect that, so instead we could set an expiration here. Maybe a day or a week?